x-pack/agent/pkg/agent/program/program.go

```diff
@@ -173,10 +173,7 @@ func groupByOutputs(single *transpiler.AST) (map[string]*transpiler.AST, error)
 		grouped[k] = clone
 	}
 
-	s, ok := normMap[streamsKey]
-	if !ok {
-		return nil, errors.New("no streams are configured")
-	}
+	s, _ := normMap[streamsKey]
```

Comment:
test down below does not test this scenario. let's make sure we have a valid path in case there's nothing under `streamsKey` (casting to array fails)

Reply:
good catch.